velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -153,6 +155,82 @@ TpchPlan TpchQueryBuilder::getQ1Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ4Plan() const {
+  static const std::string kOrders = "orders";
+  static const std::string kLineitem = "lineitem";
+
+  std::vector<std::string> lineitemColumns = {
+      "l_orderkey", "l_commitdate", "l_receiptdate"};
+  std::vector<std::string> ordersColumns = {
+      "o_orderdate", "o_orderkey", "o_orderpriority"};
+
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders);
+
+  // dateVal :2 = '1993-07-01'
+  const auto orderDate = "o_orderdate";
+  std::string orderDateFilter;
+
+  // DWRF does not support Date type. Use Varchar instead.
+  if (ordersSelectedRowType->findChild(orderDate)->isVarchar()) {
+    orderDateFilter = "o_orderdate between '1993-07-01' and '1993-09-30'";
+  } else {
+    orderDateFilter =
+        "o_orderdate between '1993-07-01'::DATE and '1993-09-30'::DATE";
+  }
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId lineitemPlanNodeId;
+  core::PlanNodeId ordersPlanNodeId;
+
+  auto lineitems = PlanBuilder(planNodeIdGenerator)
+                       .tableScan(
+                           kLineitem,
+                           lineitemSelectedRowType,
+                           lineitemFileColumns,
+                           {},
+                           "l_commitdate < l_receiptdate")
+                       .capturePlanNodeId(lineitemPlanNodeId)
+                       .planNode();
+
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .localPartition(
+              {},
+              {PlanBuilder(planNodeIdGenerator)
+                   .tableScan(
```

Comment:
Again a question about join order. Build side (lineitem) appears to be larger than probe side (orders). For sf1: 37897 vs. 582 rows. It seems like it would make sense to swap. In offline discussion, you mentioned that you replaced semi join with 'group by => right outer join => NOT NULL filter'. Do you plan to update this PR to reflect that change?

Reply:
Yes, I will be updating the PR with this change, I wanted to take all the comments and push the changes together, I will push these changes shortly